### PR TITLE
Add Life List page: every confirmed species with its best photo

### PR DESCRIPTION
**Pull request by jss367**
## What changed

New **Life List** page (`/life-list`) — every species the user has confirmed at least one photo of, shown as a card grid:

- **Lifer numbering**: species are numbered chronologically by first-photographed date, the way birders count lifers. Species with no capture time sort last (alphabetically) so numbering stays stable.
- **Best photo per species**: each card's hero image is picked by the existing highlights scorer (`_highlight_score_bucket`), so eye focus / exposure / composition and user ratings all factor in. Clicking a card opens the lightbox over the species' top photos.
- **Card details**: photo count, first/latest photographed dates, scientific name in italics when the keyword is linked to a taxon, and location-keyword chips.
- **Controls**: client-side species search (matches keyword, scientific, and common names) and sort (newest lifer / life order / A–Z / most photos).

### Semantics

- **Confirmed species only** — a prediction alone doesn't put a species on your life list (AI suggests, humans decide). Species keywords are recognized the same way as everywhere else (`is_species = 1` OR `type = 'taxonomy'`).
- **Unscored photos still count** — unlike Highlights, photos with no `quality_score` are included, so a species confirmed before any pipeline run isn't dropped from the list. The scorer falls back gracefully.
- Rejected photos and folders outside the active workspace are excluded.

### Implementation

- `vireo/db.py`: `get_life_list_candidates()` (photo × species-keyword rows with quality metrics + taxa names) and `get_life_list_locations()`; `life_list` added to `ALL_NAV_IDS`.
- `vireo/app.py`: `/life-list` page route, `/api/life-list` endpoint, `ALL_PAGES` entry.
- `vireo/templates/_navbar.html`: `NAV_ALL_PAGES` fallback kept in sync.
- `vireo/templates/life_list.html`: new page, modeled on highlights.html conventions (escaped string templates, vireo-base.css variables, shared lightbox).
- Not added to `DEFAULT_TABS` — reachable via the palette/overflow like Map or Dashboard.

## Tests

- New `vireo/tests/test_life_list.py` (8 tests): grouping/counts, best-photo ranking, lifer numbering and first/last dates, taxon names, location chips, unscored-photo inclusion, workspace scoping, page render.
- Updated the hardcoded `ALL_NAV_IDS` parity test in `test_db.py`.
- Full standard suite: **1089 passed, 1 skipped**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added "Life List" page to navigation with a searchable, sortable species grid and thumbnails.
  * Species cards show first/last seen dates, up to three location chips, photo counts, and open a lightbox of best photos.
  * Added sort options: newest, life-order/oldest-first, A–Z, and most-photos.

* **Tests**
  * Expanded test coverage for Life List behavior and edge cases (null flags, duplicate-named species tags).

### CODE REVIEW

**Comment by jss367** [comment]
🤖 CI failure on run [27314383448](https://github.com/jss367/vireo/actions/runs/27314383448) was infrastructure flakiness, not a code issue — the `test` job's runner received a shutdown signal at ~97% completion (`##[error]The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled.`). Every test up to that point passed and the `lint` job was clean.

No code changes needed. Re-ran the failed job rather than pushing a no-op commit.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: vireo/db.py [hunk 1]
```diff
@@ -7202,6 +7202,77 @@ def get_highlights_candidates(self, folder_id, min_quality=0.0):
...
+               FROM photo_keywords pk
+               JOIN keywords k ON k.id = pk.keyword_id
+                AND (k.is_species = 1 OR k.type = 'taxonomy')
+               JOIN photos p ON p.id = pk.photo_id
+               JOIN workspace_folders wf ON wf.folder_id = p.folder_id
+                AND wf.workspace_id = ?
+               JOIN folders f ON f.id = p.folder_id
+                AND f.status IN ('ok', 'partial')
+               LEFT JOIN taxa t ON t.id = k.taxon_id
+               WHERE p.flag != 'rejected'
```
[the remark this answers is not in the document]
Comment (jss367) [reply vireo/db.py]: Fixed in 7b83ade2 — both life-list queries now use `COALESCE(p.flag, 'none') != 'rejected'`, matching the NULL-safe flag handling used elsewhere in db.py. Added a regression test (`test_null_flag_photo_included`) seeding NULL-flag photos through both the candidates and locations queries.

File: vireo/db.py [hunk 2]
```diff
@@ -7202,6 +7202,77 @@ def get_highlights_candidates(self, folder_id, min_quality=0.0):
...
+               ORDER BY k.name, p.timestamp""",
```
[the remark this answers is not in the document]
Comment (jss367) [reply vireo/db.py]: Fixed in 7b83ade2 with `COALESCE(p.flag, 'none') != 'rejected'` on both queries, plus a regression test seeding NULL-flag photos.

File: vireo/templates/life_list.html [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    + escapeAttr((entry.locations || []).join(', ')) + '">'
+    + '<div class="lifer-number">#' + entry.number + '</div>'
+    + (best ? '<img src="' + thumb + '" alt="' + escapeAttr(best.filename) + '" loading="lazy">' : '')
+    + '<div class="species-info">'
+    + '<div class="species-name">' + escapeAttr(entry.species) + '</div>'
+    + sci
+    + '<div class="species-meta">' + metaParts.join(' &middot; ') + '</div>'
+    + (locations ? '<div class="species-locations">' + locations + '</div>' : '')
+    + '</div></div>';
+}
```
[the remark this answers is not in the document]
Comment (jss367) [reply vireo/templates/life_list.html:142]: Done in 7b83ade2 — the thumb URL now goes through `escapeAttr` (note `best.id` is an integer database ID so this is defense-in-depth rather than an exploitable path, but it keeps the convention uniform). Also added the loading/failure state from the nitpick. Skipped the `escapeHtml`-for-text-nodes refinement (self-rated low value): `escapeAttr` over-escapes but is safe and matches the highlights.html convention.

**Comment by jss367** [comment]
🤖 Received an `address-review` routine fire (author: jss367) with an empty review body. Reviewed the PR's open and resolved threads — all reviewer feedback (NULL-flag handling in `vireo/db.py`, thumbnail URL escaping + loading state in `life_list.html`) has already been addressed in commit 7b83ade2 with reply comments explaining the fixes. No outstanding requests to act on, so no commit was pushed.

If there was specific feedback you intended to convey here, please re-comment with the request body and I'll pick it up.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jss367** [comment]
@codex review

**Comment by jss367** [comment]
🤖 Received an `address-comment` routine fire for the `@codex review` comment. That's a mention routing to the Codex bot (which has already 👀-reacted), not feedback requesting code changes from me — so no commit was pushed.

If you'd like me to act on Codex's review once it posts, drop a `/claude-fix` comment after the review lands.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jss367** [comment]
🤖 Received an `address-comment` routine fire for Codex's status update. Codex's comment is itself informational — it explicitly says no code changes were made because the `@codex review` mention was a routing event, not feedback. There are no review findings, suggestions, or requested changes embedded in it, so no commit was pushed.

If Codex (or another reviewer) posts substantive feedback later, drop a `/claude-fix` comment and I'll pick it up.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jss367** [comment]
🤖 Received another `address-comment` routine fire for a Codex no-op status update — functionally identical to the one I already addressed in [4676164269](https://github.com/jss367/vireo/pull/966#issuecomment-4676164269) one minute ago. Codex's comment again explicitly states no code changes were made because the trigger was informational; there are no review findings, suggestions, or requested changes embedded in it.

No commit pushed. I won't keep responding to follow-up Codex no-op comments to avoid a notification loop — please drop a `/claude-fix` only when there's substantive reviewer feedback to act on.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jss367** [comment]
🤖 `/claude-fix` audit — re-checked every review/comment on the PR against the current branch (`7b83ade2`):

| Feedback | Status |
| --- | --- |
| Codex: NULL-flag filter on `db.py:7241,7260` | ✅ `COALESCE(p.flag, 'none') != 'rejected'` on both queries |
| CodeRabbit: NULL-flag filter (same) | ✅ Fixed + thread auto-acknowledged |
| CodeRabbit: thumbnail `src` XSS — escape `thumb` | ✅ `escapeAttr(thumb)` at `life_list.html:126` |
| CodeRabbit nit: loading/failure state | ✅ `"Loading life list…"` / `"Failed to load life list."` at `life_list.html:85,89` |
| CodeRabbit nit: `escapeHtml` for text nodes | ❎ Intentionally skipped (matches `highlights.html` convention) — CodeRabbit accepted |
| Docstring coverage warning (25%) | New DB methods both have docstrings; route handlers follow project convention (no docstring on `*_page`/`api_*`) |

Ran the relevant suite locally:

```
python -m pytest vireo/tests/test_life_list.py vireo/tests/test_db.py -q
479 passed, 2 warnings in 16.16s
```

Nothing actionable left from the existing reviews — the only outstanding item is the `@codex review` request you posted, which Codex has 👀-acked but not yet returned. Drop another `/claude-fix` once that review lands and I'll address whatever it surfaces.

---
_Generated by [Claude Code](https://claude.ai/code)_
